src/test/java/seedu/address/model/LeavesBookTest.java

```diff
@@ -0,0 +1,185 @@
+package seedu.address.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalLeaves.ALICE_LEAVE;
+import static seedu.address.testutil.TypicalLeaves.BOB_LEAVE;
+import static seedu.address.testutil.TypicalLeaves.getTypicalLeavesBook;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.leave.Leave;
+import seedu.address.model.leave.exceptions.DuplicateLeaveException;
+import seedu.address.model.leave.exceptions.LeaveNotFoundException;
+import seedu.address.testutil.LeaveBuilder;
+
+public class LeavesBookTest {
+
+    private final LeavesBook leavesBook = new LeavesBook();
+
+    @Test
+    public void constructor() {
+        assertEquals(Collections.emptyList(), leavesBook.getLeaveList());
+    }
+
+    @Test
+    public void resetData_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> leavesBook.resetData(null));
+    }
+
+    @Test
+    public void resetData_withValidReadOnlyLeavesBook_replacesData() {
+        LeavesBook newData = getTypicalLeavesBook();
+        leavesBook.resetData(newData);
+        assertEquals(newData, leavesBook);
+    }
+
+    @Test
+    public void resetData_withDuplicateLeaves_throwsDuplicateLeaveException() {
+        // Two leaves with the same identity fields
+        Leave editedAlice = new LeaveBuilder(ALICE_LEAVE).withDescription(
+                "Alice's Maternity Leave Description").build();
+        List<Leave> newLeaves = Arrays.asList(ALICE_LEAVE, editedAlice);
+        LeavesBookStub newData = new LeavesBookStub(newLeaves);
+
+        assertThrows(DuplicateLeaveException.class, () -> leavesBook.resetData(newData));
+    }
+
+    @Test
+    public void hasLeave_nullLeave_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> leavesBook.hasLeave(null));
+    }
+
+    @Test
+    public void hasLeave_leaveNotInLeavesBook_returnsFalse() {
+        assertFalse(leavesBook.hasLeave(ALICE_LEAVE));
```

Comment:
I think it would be better to either use a new LeavesBook or implement set up and tear down for each unit test, since the unit tests should fulfil test independence and I'm not sure if the tests are executed in a deterministic order. This could be a problem if a previous test does not clean up e.g. if some previous unit test adds ALICE_LEAVE, this test case would fail.

Reply:
https://www.baeldung.com/junit-testinstance-annotation, according to the link here, it says that a new instance of the class is created when running each test method. But if it adheres to better coding practices, then i shall change it. What are your opinions on this?

Reply:
Hmm if it is a new instance each time, I think it should be ok